src/axom/sina/tests/sina_Document.cpp

```diff
@@ -0,0 +1,373 @@
+  EXPECT_EQ(records[0]->getFiles().count(File {"test/test.png"}), 1);
+}
+
+TEST(Document, toNode_empty)
+{
+  // A sina document should always have, at minimum, both records and
+  // relationships as empty arrays.
+  Document const document;
+  conduit::Node asNode = document.toNode();
+  EXPECT_TRUE(asNode[EXPECTED_RECORDS_KEY].dtype().is_list());
+  EXPECT_EQ(0, asNode[EXPECTED_RECORDS_KEY].number_of_children());
+  EXPECT_TRUE(asNode[EXPECTED_RELATIONSHIPS_KEY].dtype().is_list());
+  EXPECT_EQ(0, asNode[EXPECTED_RELATIONSHIPS_KEY].number_of_children());
+}
+
+TEST(Document, toNode_records)
+{
+  Document document;
+  std::string expectedIds[] = {"id 1", "id 2", "id 3"};
+  std::string expectedValues[] = {"value 1", "value 2", "value 3"};
+
+  auto numRecords = sizeof(expectedIds) / sizeof(expectedIds[0]);
+  for(std::size_t i = 0; i < numRecords; ++i)
+  {
+    document.add(std::make_unique<TestRecord<std::string>>(expectedIds[i],
+                                                           TEST_RECORD_TYPE,
+                                                           expectedValues[i]));
+  }
+
+  auto asNode = document.toNode();
+
+  auto record_nodes = asNode[EXPECTED_RECORDS_KEY];
+  ASSERT_EQ(numRecords, record_nodes.number_of_children());
+  for(auto i = 0; i < record_nodes.number_of_children(); ++i)
+  {
+    auto &actualNode = record_nodes[i];
+    EXPECT_EQ(expectedIds[i], actualNode["id"].as_string());
+    EXPECT_EQ(TEST_RECORD_TYPE, actualNode["type"].as_string());
+    EXPECT_EQ(expectedValues[i], actualNode[TEST_RECORD_VALUE_KEY].as_string());
+  }
+}
+
+TEST(Document, toNode_relationships)
+{
+  Document document;
+  std::string expectedSubjects[] = {"subject 1", "subject 2"};
+  std::string expectedObjects[] = {"object 1", "object 2"};
+  std::string expectedPredicates[] = {"predicate 1", "predicate 2"};
+
+  auto numRecords = sizeof(expectedSubjects) / sizeof(expectedSubjects[0]);
+  for(unsigned long i = 0; i < numRecords; ++i)
+  {
+    document.add(Relationship {
+      ID {expectedSubjects[i], IDType::Global},
+      expectedPredicates[i],
+      ID {expectedObjects[i], IDType::Global},
+    });
+  }
+
+  auto asNode = document.toNode();
+
+  auto relationship_nodes = asNode[EXPECTED_RELATIONSHIPS_KEY];
+  ASSERT_EQ(numRecords, relationship_nodes.number_of_children());
+  for(auto i = 0; i < relationship_nodes.number_of_children(); ++i)
+  {
+    auto &actualRelationship = relationship_nodes[i];
+    EXPECT_EQ(expectedSubjects[i], actualRelationship["subject"].as_string());
+    EXPECT_EQ(expectedObjects[i], actualRelationship["object"].as_string());
+    EXPECT_EQ(expectedPredicates[i], actualRelationship["predicate"].as_string());
+  }
+}
+
+/**
+ * Instances of this class acquire a temporary file name when created
+ * and delete the file when destructed.
+ *
+ * NOTE: This class uses unsafe methods and should only be used for testing
+ * purposes. DO NOT move it to the main code!!!!
+ */
+class NamedTempFile
```

Comment:
Can we do this in a separate PR?

Reply:
Definitely -- I was mostly pointing out connections w/ my comments -- not requiring that they happen in this PR. The main things I'm concerned about for this PR are the SINA_VERSION defines and build system integration

Reply:
Ok great, I'm going to go through the build system stuff today. Whatever I don't end up doing in this PR will just be left as an unresolved conversation so it can be revisited in the future.